### PR TITLE
Add jdbcx driver

### LINKED ISSUE #20823: Add JDBCX driver
JDBCX is a JDBC wrapper that incorporates a lightweight template engine to enable dynamic query generation.
```
{% var: num=3 %}
select {{ script: ${num} - 2 }} one, {{ shell: echo 2 }} two, {{ sql(id=ch-play): select ${num} }} three
```

See https://github.com/jdbcx/jdbcx

### PR DESCRIPTION
**Pull request by zhicwu**
JDBCX is a JDBC wrapper that incorporates a lightweight template engine to enable dynamic query generation.

```sql
{% var: num=3 %}
select {{ script: ${num} - 2 }} one, {{ shell: echo 2 }} two, {{ sql(id=ch-play): select ${num} }} three
```

To use JDBCX, it is recommended to create a [property file](https://en.wikipedia.org/wiki/.properties) in the following locations:
* `~/.jdbcx/config.properties` on macOS/Linux
* `%HOMEPATH%\.jdbcx\config.properties` on Windows

The configuration file should contain settings like the example below:
```
jdbcx.custom.classpath=/path/to/my/jdbc/drivers
#jdbcx.codeql.cli.path=/opt/homebrew/bin/codeql
#jdbcx.prql.cli.path=~/.cargo/bin/prqlc
```

After which, database connections can be specified by replacing the `jdbc:` prefix in the URL with `jdbcx:`. If you prefer to use PRQL as the default query language instead of SQL, replace `jdbcx:` with `jdbcx:prql:` in the connection URL.

image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.generic/plugin.xml [hunk 1]
```diff
@@ -672,6 +672,24 @@
...
+                    iconBig="icons/jdbcx_icon_big.png"
+                    description="Extended JDBC driver for dynamic multi-language queries with optional bridge server for federated datasource connectivity."
+                    supportedConfigurationTypes="MANUAL,URL"
+                    webURL="https://github.com/jdbcx/jdbcx"
+                    categories="codeql,graphql,prompt,prql,rest,script,shell,sql"
+                    singleConnection="true">
+                    <file type="jar" path="maven:/io.github.jdbcx:jdbcx-driver:RELEASE" load-optional-dependencies="true" bundle="!drivers.jdbcx" />
+                    <file type="license" path="https://www.apache.org/licenses/LICENSE-2.0.txt" bundle="!drivers.jdbcx"/>
+                    <file type="jar" path="drivers/jdbcx" bundle="drivers.jdbcx"/>
+                    <file type="license" path="drivers/apache_lic/LICENSE.txt" bundle="drivers.jdbcx"/>
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.generic/plugin.xml]: I suppose this lines should be removed as no driver and license were added to the drivers repo. We add it only for drivers that we package with DBeaver in PRO versions.

**Review by E1izabeth — APPROVED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Comment by uslss** [comment]
Hi @zhicwu , unfortunately I cannot connect to any database.
1. I've created folder with my jdbc drivers in it.
 
image

2. I've created config.properties file with custom classpath
image

3. Classpath is shown in Drivers properties in dbeaver
image

4. When I try to connect to db, I get the error below
image

Am I doing something wrong?

**Comment by zhicwu** [comment]
Hi @uslss, thanks for trying out. Which release were you using? Just tried on a windows box using 0.2.1 and I was unable to reproduce the issue.

**Comment by uslss** [comment]
@zhicwu if you talk about os version, win10 22h2 19045.3086

**Comment by zhicwu** [comment]
> @zhicwu if you talk about os version, win10 22h2 19045.3086

Thanks. I meant version of the jdbcx-driver you're using. Could you try the most recent one like v0.2.1?

Edit: is it recommended to specify driver version in plugin.xml? Let me know and I can update it to 0.2.3.

**Comment by zhicwu** [comment]
> @zhicwu if you talk about os version, win10 22h2 19045.3086

Thanks. I meant version of the jdbcx-driver you're using. Could you try the most recent one like v0.2.1?

**Comment by uslss** [comment]
Unfortunately, it is reproducible for me both against 0.2.1 and 0.2.3

**Comment by zhicwu** [comment]
> Unfortunately, it is reproducible for me both against 0.2.1 and 0.2.3

I tried on Windows 11(home edition), so I probably need to find a Windows 10 😅 Do you have a nightly build of DBeaver for me to test? Could you kindly review/try the items below when you have a moment?
1. ensure jdbcx jar does not exist in global classpath and `C:/my-jdbc-drivers`
2. the connection error should start with `Connection not established due to missing driver.`, which is slightly different from above screenshot
3. try connecting to a different database, for example: put [Derby embedded](https://repo1.maven.org/maven2/org/apache/derby/derby/10.14.2.0/derby-10.14.2.0.jar) in `C:/my-jdbc-drivers`, create a new connection using URL `jdbcx:derby:memory:x;create=true`
4. if you succeeded in above step, double check the driver version by issue query `{{ version }}`, which should return `0.2.3 (revision: 14a7179)`

**Comment by zhicwu** [comment]
Hi @uslss, could you upgrade to v0.3.0? With that, you can change connection url to `jdbcx:` and issue query `{{ help }}` to get started.

**Comment by E1izabeth** [comment]
I'll check, thank you for guidance.

**Comment by zhicwu** [comment]
Thank you @E1izabeth. One more thing, once #20969 is merged, I think we need to update this PR to set the new property to true.

**Comment by E1izabeth** [comment]
Yeah, I'm keeping it in my mind.
I'm succeeded with connecting through jdbcx driver.

**Review by E1izabeth — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Review by ShadelessFox — APPROVED** [review]

**Pull request merged**